mesheryctl/internal/cli/root/relationships/relationship.go

```diff
@@ -0,0 +1,158 @@
+// Copyright 2024 Layer5, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package relationships
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+
+	"github.com/layer5io/meshery/mesheryctl/internal/cli/root/config"
+	"github.com/layer5io/meshery/mesheryctl/internal/cli/root/system"
+	"github.com/layer5io/meshery/mesheryctl/pkg/utils"
+	"github.com/layer5io/meshkit/models/meshmodel/core/v1alpha1"
+	"github.com/manifoldco/promptui"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+var (
+	outFormatFlag string
+
+	availableSubcommands = []*cobra.Command{ViewRelationshipsCmd}
+)
+
+var RelationshipCmd = &cobra.Command{
+	Use:   "relationships",
+	Short: "View list of relationships and detailed information of a specific relationship",
+	Long:  "View list of relationships and detailed information of a specific relationship",
+	Example: `
+// To view list of components
+mesheryctl exp relationships list
+
+// To view a specific model
+mesheryctl exp relationships view [model-name]
+	`,
+	PreRunE: func(cmd *cobra.Command, args []string) error {
+		//Check prerequisite
+
+		mctlCfg, err := config.GetMesheryCtl(viper.GetViper())
+		if err != nil {
+			return utils.ErrLoadConfig(err)
+		}
+		err = utils.IsServerRunning(mctlCfg.GetBaseMesheryURL())
+		if err != nil {
+			utils.Log.Error(err)
+			return err
+		}
+		ctx, err := mctlCfg.GetCurrentContext()
+		if err != nil {
+			utils.Log.Error(system.ErrGetCurrentContext(err))
+			return nil
+		}
+		err = ctx.ValidateVersion()
+		if err != nil {
+			utils.Log.Error(err)
+			return nil
+		}
+		return nil
+	},
+	Args: func(cmd *cobra.Command, args []string) error {
+		if len(args) == 0 {
+			return cmd.Help()
+		}
+		return nil
+	},
+	RunE: func(cmd *cobra.Command, args []string) error {
+		if ok := utils.IsValidSubcommand(availableSubcommands, args[0]); !ok {
+			return errors.New(utils.SystemModelSubError(fmt.Sprintf("'%s' is an invalid subcommand. Please provide required options from [view]. Use 'mesheryctl exp relationships --help' to display usage guide.\n", args[0]), "model"))
+		}
+		_, err := config.GetMesheryCtl(viper.GetViper())
+		if err != nil {
+			return utils.ErrLoadConfig(err)
+		}
+
+		err = cmd.Usage()
+		if err != nil {
+			utils.Log.Error(err)
```

Comment:
Will this lead to two same messages being logged on the screen?
One from the log and one from the error that is returned.
May be Log.Debug?

Reply:
Yes. We'll get two same messages. I'll update.

Reply:
By using Log.Debug we are also getting the same output.
![Screenshot 2024-04-01 201050](https://github.com/meshery/meshery/assets/107461226/3c4e4beb-10c4-4aea-8230-6f9aac586c02)

Reply:
I've updated the PR. Now we are logging the errors and returning `nil`